### PR TITLE
bookmarker -p FOLDER

### PR DESCRIPTION
**Pull request by alexandre-mbm**
You may set this one time only.

### CODE REVIEW

File: lib/bookmarker/cli.ex [hunk 1]
```diff
@@ -9,6 +9,7 @@ defmodule Bookmarker.CLI do
       description: :string,
       timestamp: :boolean,
       ignore: :keep,
+      path: :string,
```
Comment (lubien) [inline lib/bookmarker/cli.ex]: Indent

File: lib/bookmarker/cli.ex [hunk 2]
```diff
@@ -17,6 +18,7 @@ defmodule Bookmarker.CLI do
       t: :title,
       d: :description,
       i: :ignore,
+      p: :path,
```
Comment (lubien) [inline lib/bookmarker/cli.ex]: Indent

File: lib/bookmarker/cli.ex [hunk 3]
```diff
@@ -50,6 +52,8 @@ defmodule Bookmarker.CLI do
             Keyword.get(params, :timestamp, true),
           ignore:
             Keyword.get_values(params, :ignore),
+          path:
```
Comment (lubien) [inline lib/bookmarker/cli.ex]: Indent

File: lib/bookmarker/cli.ex [hunk 4]
```diff
@@ -69,6 +73,7 @@ defmodule Bookmarker.CLI do
       --no-timestamp        Prevent appending of build datetime after description.
                             Default: use timestamp
       -i, --ignore          Ignore folders. You may set this multiple times.
+      -p, --path            Restrict to folder. You may set this one time only.
```
Comment (lubien) [inline lib/bookmarker/cli.ex]: Indent

File: lib/bookmarker/runner.ex [hunk 5]
```diff
@@ -5,6 +5,7 @@ defmodule Bookmarker.Runner do
   def run(config) do
     get_bookmarks(config.file)
     |> ignore_paths(config.ignore)
+    |> restrict_to(config.path)
```
Comment (lubien) [inline lib/bookmarker/runner.ex]: Indent

File: lib/bookmarker/runner.ex [hunk 6]
```diff
@@ -25,6 +26,10 @@ defmodule Bookmarker.Runner do
     end)
   end
 
+  def restrict_to(bookmarks, path) do
```
Comment (lubien) [inline lib/bookmarker/runner.ex]: Indent

File: lib/bookmarker/cli.ex [hunk 7]
```diff
@@ -50,6 +52,8 @@ defmodule Bookmarker.CLI do
             Keyword.get(params, :timestamp, true),
           ignore:
             Keyword.get_values(params, :ignore),
+          path:
+            Keyword.get_values(params, :path),
```
Comment (lubien) [inline lib/bookmarker/cli.ex]: Since you use only one path, why get multiple values?

**Comment by lubien** [comment]
Rebase your branch so you'll have the Travis Fix

File: mix.exs [hunk 8]
```diff
@@ -3,7 +3,7 @@ defmodule Bookmarker.Mixfile do
 
   def project do
     [app: :bookmarker,
-     escript: escript_config,
+     escript: escript_config(),
```
Comment (lubien) [inline mix.exs]: Leave this issue to the other pull request. Please undo this commit.

Comment (alexandre-mbm) [inline mix.exs]: With no rebase it would have the same hash and would be the same of fact.

File: lib/bookmarker/cli.ex (hunk 1 above)
Comment (alexandre-mbm) [inline lib/bookmarker/cli.ex]: But it is OK

File: lib/bookmarker/cli.ex (hunk 2 above)
Comment (alexandre-mbm) [inline lib/bookmarker/cli.ex]: But it is OK

File: lib/bookmarker/cli.ex (hunk 3 above)
Comment (alexandre-mbm) [inline lib/bookmarker/cli.ex]: But it is OK

File: lib/bookmarker/cli.ex (hunk 7 above)
Comment (alexandre-mbm) [inline lib/bookmarker/cli.ex]: True. I will test this.

File: lib/bookmarker/runner.ex (hunk 5 above)
Comment (alexandre-mbm) [inline lib/bookmarker/runner.ex]: But it is OK

File: lib/bookmarker/runner.ex (hunk 6 above)
Comment (alexandre-mbm) [inline lib/bookmarker/runner.ex]: But it is OK

File: mix.exs (hunk 8 above)
Comment (alexandre-mbm) [inline mix.exs]: I did rebase from new master and this commit will be right into tree.

**Comment by lubien** [comment]
So far so good but the escript fix bothers me. Any way to remove that commit?

**Comment by alexandre-mbm** [comment]
At result merged you will not see difference. The commit is exactly the same, with the same hash. **It will is not duplicated.** You can trust! **Nothing will turns dirty.** Please. It will is a _fast forward_ merging, I think.

**Comment by alexandre-mbm** [comment]
My experiment:
```console
$ git merge fork/dir 
Merge made by the 'recursive' strategy.
 lib/bookmarker/bookmark.ex | 7 +++++++
 lib/bookmarker/cli.ex      | 5 +++++
 lib/bookmarker/runner.ex   | 5 +++++
 3 files changed, 17 insertions(+)
```
```console
$ git tree 
*   8661d10 (HEAD -> master-origin) Merge remote-tracking branch 'fork/dir' into master-origin
|\  
| * 46542f4 (fork/dir, another) Fixes the semantic for only one path
| * 4f06df2 Restrict to folder
* |   92167d0 (origin/master) Merge pull request #9 from alexandre-mbm/master
|\ \  
| |/  
| * 22de841 (fork/master, master) "escript_config()" fix
|/  
*   6737df0 Merge pull request #11 from alexandre-mbm/travis
|\  
| * d25568b (fork/travis, travis) elixir 1.4.2 on travis
| * a9dde8e test for travis-ci/travis-ci#4757
|/  
* 2ffe494 Pattern matching over if..else in timestamp render
* 1e7889e Separate single bookmark rendering logic
* c0a6bae Minor updates
* c80386f Minor updates
```
I.e. all is okay!

**Comment by alexandre-mbm** [comment]
Or you have as "personal criterion" avoid non-fast-forward merging?

**Comment by alexandre-mbm** [comment]
If this web GUI does not make the _recursive_ strategy, and makes the _fast-forward_ only, so I could push my merge (no conflicts, no duplication). Do you want it?

**Comment by alexandre-mbm** [comment]
Really... if I do that, your Git history will be dirty with an unnecessary second merge.

Well, if this web GUI does not make recursive merge, or if you insist, so I will erase the commit.

**Comment by alexandre-mbm** [comment]
Don't you believe me? Am I wrong?

Note that lacks tests for the new feature. Because of this I am not in a hurry. Have you requeriments to accept the pull request?

**Comment by lubien** [comment]
It was not about trust. It was about that I had minimal access to computers, just that.

You did great. I didn't knew also that since they're the same commit it would have no problem.

**Comment by alexandre-mbm** [comment]
The same commit on the same base, being the branch previously merged or rebased.

**Pull request merged**